internal/token.go

```diff
@@ -1674,9 +1674,9 @@ expression_loop:
 				z.tt = TextToken
 				return z.tt
 			}
-			z.openBraceIsExpressionStart = true
 			z.expressionStack[len(z.expressionStack)-1]--
 			if z.expressionStack[len(z.expressionStack)-1] == -1 {
+				z.openBraceIsExpressionStart = true
```

Comment:
Pretty simple fix! This edge case exposed that `z.openBraceIsExpressionStart = true` should only be reset when we've exhausted the entire `z.expressionStack`